conftest.py

```diff
@@ -190,6 +197,38 @@ def running_linux_vm(imported_vm):
     return vm
     # no teardown
 
+@pytest.fixture(scope='session')
+def sr_deviceconfig(request):
+    raw_config = request.param
+
+    if raw_config is None:
+        # Use defaults
+        return None
+
+    config = {}
+    for key_val in raw_config.split(','):
+        key = key_val.split(':')[0]
+        value = key_val.split(':')[1:]
+        config[key] = value
+    return config
+
+@pytest.fixture(scope='session')
+def cephfs_deviceconfig(sr_deviceconfig):
+    if sr_deviceconfig is not None:
+        # SR device config from CLI param
+        config = sr_deviceconfig
+    else:
+        # SR device config from data.py defaults
+        try:
+            from data import DEFAULT_CEPHFS_DEVICECONFIG
+        except ImportError:
+            DEFAULT_CEPHFS_DEVICECONFIG = {}
+        if DEFAULT_CEPHFS_DEVICECONFIG:
+            config = DEFAULT_CEPHFS_DEVICECONFIG
+        else:
+            raise Exception("No default CephFS deviceconfig found, neither in CLI nor in data.py defaults")
+    return config
```

Comment:
Shouldn't this be in a ceph dedicated file?

Reply:
We don't know yet exactly how tests will be hierarchised and fixtures shared and there are no ceph test yet (well there are, but in a separate WIP PR that will need this), so I put it in the main conftest for now.